Makefile

```diff
@@ -26,7 +25,7 @@ publish: assembly ## Publishes artifacts on Databricks dbfs
 
 .PHONY: deploy
 deploy: publish ## Deploys artifacts, notebooks and jobs to Databricks [make deploy env=staging]
-	bash scripts/deploy_jobs.sh $(env) "$(JAR_VERSION)-$(GIT_REVISION)"
+	bash scripts/deploy_jobs.sh --environment=$(env) --build_version="$(JAR_VERSION)-$(GIT_REVISION)" --skip_jobs="$(skip_jobs)"
```

Comment:
Yes, that makes sense. I'll split them up!

Reply:
Hey, took a while, but I took another stab. 
I've changed it and split it up, and did a rebase. 
I've also extracted some common code into a separate function, so we don't have to retype the code every time. 
I do suspect it might cause some merge conflicts with the other open PR though. Let me know what you think, I can also resolve the conflicts if you want to merge the other first